### PR TITLE
initial add upstream nexenio

### PR DESCRIPTION
**Pull request by Moep90**
- [x] Made namespace configurable
- [x] Added annotations to Namespace
- [x] Added CertManagerCertificate
- [x] Added CertManagerClusterIssuer
- [x] Added CertManagerIssuer
- [x] Added host_pid for all known deployment types
- [x] Added host_network for all known deployment types
- [x] Added StartupProbe
- [x] Added/Readded VerticalPodAutoscaler

### CODE REVIEW

**Comment by Moep90** [comment]
- [x] Addded template files for everyone to copy for quickstart
- [x] Added Helmchart example for usage with kapitan
- [x]  Added example usage for cert-manager-classes

File: components/generators/kubernetes/__init__.py [hunk 1]
```diff
@@ -1013,7 +1100,8 @@ def body(self):
         super().body()
         component = self.kwargs.component
         workload = self.kwargs.workload
-        self.add_namespace(inv.parameters.namespace)
+        if component.namespace or inv.parameters.namespace:
```
Comment (ademariag) [inline components/generators/kubernetes/__init__.py:1103]: why not
```
self.add_namespace(component.get("namespace", inv.parameters.namespace))
```

File: components/generators/kubernetes/__init__.py [hunk 2]
```diff
@@ -947,7 +1008,8 @@ def body(self):
...
```
Comment (ademariag) [inline components/generators/kubernetes/__init__.py:1011]: same

File: components/generators/kubernetes/__init__.py (hunk 1 above)
Comment (ademariag) [reply components/generators/kubernetes/__init__.py:1103]: also in other places

File: components/generators/kubernetes/__init__.py [hunk 3]
```diff
@@ -419,6 +465,8 @@ def body(self):
             'restart_policy', 'Always')
         if 'host_network' in component:
             self.root.spec.template.spec.hostNetwork = component.host_network
+        if 'host_pid' in component:
```
Comment (ademariag) [inline components/generators/kubernetes/__init__.py:468]: is this duplicated?

**Review by Moep90 — COMMENTED** [review]
- [x] fixed namespace

**Review by ademariag — APPROVED** [review]

**Pull request merged**